.circleci/config.yml

```diff
@@ -16,7 +16,7 @@ commands:
 jobs:
   lint-build-debug:
     docker:
-      - image: ubuntu:23.04
+      - image: ubuntu:24.04
```

Comment:
These are the most important changes, the rest is mostly fixing errors that occurred as a result of updating the runners to 24.04.